catlearn/relation_cache.py

```diff
@@ -241,8 +313,18 @@ def match(self, labels: DirectedGraph[NodeType]) -> Tsor:
             if (
                     not self.graph.has_edge(src, tar)
                     or not self.graph[src][tar]):
+
                 for arr in self.label_universe:
-                    self.add(CompositeArrow([src, tar], [arr]))
+                    # add arrow
+                    new_arr = CompositeArrow([src, tar], [arr])
+                    self.add(new_arr)
+
+                    # match new arrow against negative label
```

Comment:
Add only if matching negatives?

Reply:
Why?

Reply:
Oh, what is after the comment..  Indeed, Sorry about that
(Was looking at what was before it so I didn't get it)